CMakeLists.txt

```diff
@@ -193,6 +193,49 @@ if(USE_PROF_API EQUAL 1)
     endif()
 endif()
 
+################
+# Detect Code Object Manager
+################
+
+if(HIP_PLATFORM STREQUAL "hcc")
+    if(DEFINED ENV{CODE_OBJ_HEADER_DIR})
+        set(CODE_OBJ_HEADER_DIR $ENV{CODE_OBJ_HEADER_DIR})
+    endif()
+    if(DEFINED ENV{CODE_OBJ_MGR_DIR})
+        set(CODE_OBJ_MGR_DIR $ENV{CODE_OBJ_MGR_DIR})
+    endif()
+
+    find_path(CODE_OBJ_HEADER amd_comgr.h
+        HINTS
+            ${CODE_OBJ_HEADER_DIR}
+        PATHS
+            /opt/rocm/include
+    )
+    if(NOT CODE_OBJ_HEADER)
+        MESSAGE(WARNING "Code Object Manager header not found. Use -DCODE_OBJ_HEADER_DIR=<path_to_amd_comgr.h>.")
+    endif()
+
+    find_library(CODE_OBJ_MGR amd_comgr
+        HINTS
+            ${CODE_OBJ_MGR_DIR}
+        PATHS
+            /opt/rocm/lib
+    )
+    if (NOT CODE_OBJ_MGR)
+        MESSAGE(FATAL_ERROR "Code Object Manager not found. Use -DCODE_OBJ_MGR_DIR=<path_to_libamd_comgr.so>.")
+    else()
+        include_directories(${CODE_OBJ_HEADER})
+        add_library(amd_comgr SHARED IMPORTED GLOBAL)
+        set_target_properties(amd_comgr PROPERTIES
+            INTERFACE_INCLUDE_DIRECTORIES "${CODE_OBJ_HEADER}"
```

Comment:
Comgr should already set this in amd_comgr-config.cmake; I don't think most of the changes here are required, as just `find_library` with an optional hint should be enough. If that doesn't work, I would consider it a bug in Comgr.

Reply:
I've tried to do that, but ran into some issue. I believe that should be dealt with in a separate PR.

Reply:
Hi Scott, I've updated the CMAKE to use the find_package instead

Reply:
LGTM, thank!